crates/miden-tx/src/auth/tx_authenticator.rs

```diff
@@ -213,11 +234,25 @@ impl TransactionAuthenticator for BasicAuthenticator {
 
         async move {
             match self.keys.get(&pub_key_commitment) {
-                Some(key) => Ok(key.sign(message)),
+                Some((auth_key, _)) => Ok(auth_key.sign(message)),
                 None => Err(AuthenticationError::UnknownPublicKey(pub_key_commitment)),
             }
         }
     }
+
+    /// Returns the public key associated with the given public key commitment.
+    ///
+    /// Supported signature schemes:
+    /// - RpoFalcon512
+    /// - EcdsaK256Keccak
+    ///
+    /// If the public key commitment is not contained in the `keys` map, `None` is returned.
+    fn get_public_key(
+        &self,
+        pub_key_commitment: PublicKeyCommitment,
+    ) -> impl FutureMaybeSend<Option<PublicKey>> {
+        async move { self.keys.get(&pub_key_commitment).map(|(_, pub_key)| pub_key.clone()) }
+    }
```

Comment:
I would maybe change the return type from `PublicKey` to `&PublicKey` to avoid cloning.